confgenerator/logging_processors.go

```diff
@@ -523,3 +523,28 @@ func (p LoggingProcessorExcludeLogs) Processors(ctx context.Context) ([]otel.Com
 func init() {
 	LoggingProcessorTypes.RegisterType(func() LoggingProcessor { return &LoggingProcessorExcludeLogs{} })
 }
+
+// LoggingMultiProcessorMixin implements the methods required to describe a pipeline with one or more log processors.
+type LoggingMultiProcessorMixin interface {
+	Type() string
+	// Processors returns slice of logging processors. This is an intermediate representation before sub-agent specific configurations.
+	Processors(ctx context.Context) []LoggingProcessorMixin
+}
+
+// LoggingMultiProcessor represents a pipeline that consists of one or more log processors.
+type LoggingMultiProcessor[P LoggingMultiProcessorMixin] struct {
```

Comment:
Let's see if we can make this an unexported internal type (see comment in flink.go)

Reply:
This became `loggingProcessorMacroAdapter`.